src/explore-education-statistics-common/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -61,6 +62,54 @@ function getExcludedFilters(
   );
 }
 
+/**
+ * Determines whether any rows or columns are excluded from the table because they have no data.
+ * For filters, indicators and locations:
+ *  - when there's no data, these aren't included in subjectMeta.
+ *  - compare subjectMeta with the query to find the excluded ones.
+ *  For timePeriod:
+ *  - when there's no data, these are in subjectMeta but aren't in tableHeadersConfig.
+ *  - compare subjectMeta with tableHeadersConfig to find the excluded ones.
+ *  - the query isn't useful here as just has the start and end timePeriods.
+ */
+const hasExcludedRowsOrColumns = (
+  query: ReleaseTableDataQuery,
+  subjectMeta: FullTableMeta,
+  tableHeadersConfig: TableHeadersConfig,
+): boolean => {
+  if (
+    query.locationIds.length !== subjectMeta.locations.length ||
+    query.indicators.length !== subjectMeta.indicators.length
+  ) {
+    return true;
+  }
+
+  const subjectMetaFilters = Object.values(subjectMeta.filters)
+    .flatMap(filterGroup => filterGroup.options)
+    .map(filter => filter.id);
+
+  if (query.filters.length !== subjectMetaFilters.length) {
+    return true;
+  }
+
+  const tableHeaderFilters = [
+    ...tableHeadersConfig.columnGroups.flatMap(filterGroup => filterGroup),
+    ...tableHeadersConfig.rowGroups.flatMap(filterGroup => filterGroup),
+    ...tableHeadersConfig.columns,
+    ...tableHeadersConfig.rows,
+  ].map(filter => filter.id);
```

Comment:
I'll have a proper look next week, but I'm not sure that would work as excluded filters currently includes ones that aren't shown in the table because there's only one option, so it applies to everything in the table rather being an individual row/column (I'm not explaining that very well!).

Reply:
I've extracted `tableHeaderFilters` out and pass it in to both functions now to reduce the duplication. As mentioned above `excludedFilters` is different so it can't be used for showing the warning. I've updated the name of the new function to `excludesRowsOrColumnsWithNoData` to try to make the distinction clearer.

I've changed `getExcludedFilters` to return a Set as suggested.